modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -335,10 +352,35 @@ public static void startHouseKeepingThreads() throws AndesException {
 
         //reload exchanges/queues/bindings and subscriptions
         AndesRecoveryTask andesRecoveryTask = new AndesRecoveryTask();
-        Integer scheduledPeriod = AndesConfigurationManager.readValue
+        //deleted the expired message from db
+        PeriodicExpiryMessageDeletionTask periodicExpiryMessageDeletionTask = null;
+
+        int recoveryTaskScheduledPeriod = AndesConfigurationManager.readValue
                 (AndesConfiguration.PERFORMANCE_TUNING_FAILOVER_VHOST_SYNC_TASK_INTERVAL);
-        andesRecoveryTaskScheduler.scheduleAtFixedRate(andesRecoveryTask, scheduledPeriod,
-                                                       scheduledPeriod, TimeUnit.SECONDS);
+        int dbBasedDeletionTaskScheduledPeriod = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_MESSAGE_EXPIRATION_CHECK_INTERVAL);
+        int safeDeleteRegionSlotCount = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_SAFE_DELETE_REGION_SLOT_COUNT);
+        boolean isDLCExpiryCheckEnabled = AndesConfigurationManager.readValue
+                (AndesConfiguration.PERFORMANCE_TUNING_ALLOW_EXPIRATION_CHECK_IN_DLC);
+
+        //based on the DLC expiration check configuration bind the appropriate deletion task
+        if(isDLCExpiryCheckEnabled){
```

Comment:
We need to do it in separate places since one is associated with messaging engine initialization and the other is a starting a housekeeping thread